result/src/main/kotlin/com/github/kittinunf/result/Result.kt

```diff
@@ -4,59 +4,96 @@ package com.github.kittinunf.result
  * Created by Kittinun Vantasin on 10/26/15.
  */
 
-public interface ResultType<out V : Any, out E : Exception> {
```

Comment:
I noticed indeed but I think it doesn't add much value in general to be able to define a own concrete type because the result type interface already defines where it is always a 2 state type. So it always becomes some specification of the general Result implementation, so the same specification can already be achieved with the generic type params. Extension functions can cope with any addition you wish to make to the standard type. So therefor I think being able to extent does not add that much that justifies this design.